consensus/phases/data.go

```diff
@@ -0,0 +1,203 @@
+/*
+ *    Copyright 2018 Insolar
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package phases
+
+import (
+	"crypto/ecdsa"
+
+	"github.com/insolar/insolar/core"
+)
+
+type PacketType uint8
+type ClaimType uint8
+type ReferendumType uint8
+
+const (
+	Type1 = PacketType(iota + 1)
+)
+
+const (
+	TypeNodeClaim = ClaimType(iota + 1)
+	TypeNodeViolationBlame
+	TypeCapabilityPollingAndActivation
+	TypeNodeBroadcast
+)
+
+// ----------------------------------PHASE 1--------------------------------
+
+type PacketHeader struct {
+	Routing      uint8
+	Pulse        uint32
+	OriginNodeID uint32
+	TargetNodeID uint32
+}
+
+// PulseDataExt is a pulse data extension.
+type PulseDataExt struct {
+	NextPulseDelta uint16
+	PrevPulseDelta uint16
+	OriginID       uint16
+	EpochPulseNo   uint32
+	PulseTimestamp uint32
+	Entropy        core.Entropy
+}
+
+// PulseData is a pulse data.
+type PulseData struct {
+	PulseNumer uint32
+	Data       *PulseDataExt
+}
+
+type NodePulseProof struct {
+	NodeStateHash uint64
+	NodeSignature uint64
+}
+
+// --------------REFERENDUM--------------
+
+type ReferendumClaim interface {
+	Type() ClaimType
+	Length() uint16
+}
+
+// NodeBroadcast is a broadcast of info. Must be brief and only one entry per node.
+// Type 4.
+type NodeBroadcast struct {
+	EmergencyLevel uint8
+	claimType      ClaimType
+	length         uint16
+}
+
+func (nb *NodeBroadcast) Type() ClaimType {
+	return nb.claimType
```

Comment:
`return TypeNodeBroadcast` instead (and return corresponding enum values in all other `ClaimTypes`) and remove field `nb.claimType`

Reply:
fixed